src/postgres/mod.rs

```diff
@@ -70,7 +122,11 @@ impl Default for Postgres {
         env_vars.insert("POSTGRES_USER".to_owned(), "postgres".to_owned());
         env_vars.insert("POSTGRES_PASSWORD".to_owned(), "postgres".to_owned());
 
-        Self { env_vars }
+        Self {
+            env_vars,
+            init_mounts: Vec::new(),
+            tempdir: Arc::new(tempfile::tempdir().ok()),
```

Comment:
Good point. Did not think about this, moving tempdir-cretaition into `with_init_sql` also improves the error handling in `with_init_sql`..

Reply:
(lets split this into two parts and discuss if async is needed here instead of the other comment ^^)

https://docs.rs/tempfile/latest/tempfile/ does not offer an async API.
This may be because of two reasons:
- [nobody has requested it yet](https://github.com/Stebalien/tempfile/issues?q=is%3Aissue+async+)
- creating a directory is realy quick => having the async around this might be slower due to the overhead.

Also, since AsyncDrop is not a thing... https://github.com/sunsided/async-tempfile-rs/issues/1 any implemntation needs to be sync anyway (at least for the drop)

Reply:
I don't suggest using async api of tempfile. I'm aware it doesn't provide this. But it's not recommended to use blocking calls within async runtime. So it would be nice to cover with `spawn_blocking`, but not necessary on current stage - we could just declare that the api is blocking to warn users

>     creating a directory is realy quick => having the async around this might be slower due to the overhead.

It's actually not really true, it depends on many factors. Highly depends on disk usage and file-system. 

Moreover, we not only create a directory, but also write a file

Reply:
good point